flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/expressions/ScalarFunctionsTest.scala

```diff
@@ -4195,4 +4195,53 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
+  @Test
+  def testJsonQuery(): Unit = {
+    // lax test
+    testSqlApi("json_query('{\"foo\":100}', 'lax $' null on empty)", "{\"foo\":100}")
+    testSqlApi("json_query('{\"foo\":100}', 'lax $' error on empty)", "{\"foo\":100}")
+    testSqlApi("json_query('{\"foo\":100}', 'lax $' empty array on empty)", "{\"foo\":100}");
+    testSqlApi("json_query('{\"foo\":100}', 'lax $' empty object on empty)", "{\"foo\":100}");
+    testSqlApi("json_query('{\"foo\":100}', 'lax $.foo' null on empty)", "null");
+    testSqlApi("json_query('{\"foo\":100}', 'lax $.foo' empty array on empty)", "[]");
+    testSqlApi("json_query('{\"foo\":100}', 'lax $.foo' empty object on empty)", "{}");
+
+    // path error test
+    testSqlApi("json_query('{\"foo\":100}', 'invalid $.foo' null on error)", "null");
+    testSqlApi("json_query('{\"foo\":100}', 'invalid $.foo' empty array on error)", "[]");
+    testSqlApi("json_query('{\"foo\":100}', 'invalid $.foo' empty object on error)", "{}");
+
+    // strict test
+    testSqlApi("json_query('{\"foo\":100}', 'strict $' null on empty)", "{\"foo\":100}");
+    testSqlApi("json_query('{\"foo\":100}', 'strict $' error on empty)", "{\"foo\":100}");
+    testSqlApi("json_query('{\"foo\":100}', 'strict $' empty array on error)", "{\"foo\":100}");
+    testSqlApi("json_query('{\"foo\":100}', 'strict $' empty object on error)", "{\"foo\":100}");
+
+    testSqlApi("json_query('{\"foo\":100}', 'strict $.foo1' null on error)", "null");
+    testSqlApi("json_query('{\"foo\":100}', 'strict $.foo1' empty array on error)", "[]");
+    testSqlApi("json_query('{\"foo\":100}', 'strict $.foo1' empty object on error)", "{}");
+    testSqlApi("json_query('{\"foo\":100}', 'strict $.foo' null on error)", "null");
+    testSqlApi("json_query('{\"foo\":100}', 'strict $.foo' empty array on error)", "[]");
+    testSqlApi("json_query('{\"foo\":100}', 'strict $.foo' empty object on error)", "{}");
+
+    // array wrapper test
+    testSqlApi("json_query('{\"foo\":100}', 'strict $.foo' without wrapper)", "null");
+    testSqlApi("json_query('{\"foo\":100}', 'strict $.foo' without array wrapper)", "null");
+    testSqlApi("json_query('{\"foo\":100}', 'strict $.foo' with wrapper)", "[100]");
+    testSqlApi("json_query('{\"foo\":100}', 'strict $.foo' with unconditional wrapper)", "[100]");
+    testSqlApi("json_query('{\"foo\":100}', 'strict $.foo' with conditional wrapper)", "[100]");
+    testSqlApi("json_query('{\"foo\":[100]}', 'strict $.foo' without wrapper)", "[100]");
+    testSqlApi("json_query('{\"foo\":[100]}', 'strict $.foo' without array wrapper)", "[100]");
+    testSqlApi("json_query('{\"foo\":[100]}', 'strict $.foo' with wrapper)", "[[100]]");
+    testSqlApi("json_query('{\"foo\":[100]}', 'strict $.foo' with unconditional wrapper)",
+      "[[100]]");
+    testSqlApi("json_query('{\"foo\":[100]}', 'strict $.foo' with conditional wrapper)", "[100]");
+
+    // nulls
+    testSqlApi("json_query(cast(null as varchar), 'lax $')", "null")
```

Comment:
Please also add some tests on a column reference? Currently, all the testing data is char.
Please also test on non-string columns.